app/services/forecast_service.rb

```diff
@@ -1,16 +1,25 @@
 class ForecastService < BaseService
 
+  def self.base_url
+    "http://api.openweathermap.org"
+  end
+
+  def self.set_api_key(connection)
+    connection.params['key'] = ENV['OW_API_KEY']
+  end
+
   def self.get_city_forecast(city)
     coordinates = MapquestGeoService.get_coordinates(city)
     lat = coordinates[:results][0][:locations][0][:latLng][:lat]
     lon = coordinates[:results][0][:locations][0][:latLng][:lng]
```

Comment:
`Hash.dig` and the safety operator let you safely traverse an API response without blowing up when the data structure is unexpected.

Ruby allows you to return multiple values from a method by returning an array. Callers can then unpack that array into multiple local variables.

Resilient APIs gracefully handle unexpected results or service outages.

```
def self.get_city_forecast(city)
  ...
  lat, lon = unpack_coordinates(coordinates)

  if lat.nil? || lon.nil?
    # do some error handling
  end
  ...
end

private

def unpack_coordinates(coordinates)
  root = coordinates.dig(:results)&.first&.dig(:latLng)

  [root&.dig(:lat), root&.dig(lng)]
end
```

In the new inplementaton, a malformed or unexpected result won't raise an exception. Instead, this method would return `[nil, nil]`. That allows any error-handling code you add to the caller to make decisions about how to proceed gracefully.